extension/controller/backgroundScript.js

```diff
@@ -34,6 +40,14 @@ let translationRequestsByTab = new Map();
 let outboundRequestsByTab = new Map();
 const translateAsBrowseMap = new Map();
 
+// set data collection accordingly with what the user set
+browser.storage.local.get("telemetryCollectionConsent").then(item => {
+  if (Object.keys(item).length !== 0) {
+    settings.uploadTelemetry = item.telemetryCollectionConsent;
```

Comment:
do not change settings, it blocks sending deletion ping because PingSender thinks it's a debugging build

Reply:
The `Object.keys(item).length !== 0` approach looks odd. It kind of works because you're looking up one key, so `!== 0` means that the pref was set and the other case would mean that there was nothing saved yet.

Does this logic match with what you have in mind?
It reads prefs and defaults to false (`: false,`) when not stored.

```js
browser.storage.local.get({
  telemetryCollectionConsent: false,
  errorCollectionConsent: false,
}).then(items => {
  if (items.telemetryCollectionConsent) {
    // enable it (assuming that it was disabled by default)
  }
  if (items.errorCollectionConsent) {
    // enable it (assuming that it was disabled by default)
  }
});
```

Reply:
yep, thanks @Rob--W !